Additional CSS: Add code comments contextualising tranformStyles for clarity

andrewserong opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
<!-- In a few words, what is the PR actually doing? -->

In the AdvancedPanel for Additional CSS, add a couple of code comments clarifying that what looks like the styles being transform is actually just a transformation being used for _validation_ only, it does not transform any of the styling rules that will be output.

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->

While reviewing #56649, a few of us were tripped up by these lines. As described in this comment (https://github.com/WordPress/gutenberg/pull/56649#issuecomment-2024383614) the reviewers got to the bottom of it, but I believe it'd help to add some code comments here for clarity, and to hopefully avoid confusion further down the track.

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->

* Add code comments, following the findings in https://github.com/WordPress/gutenberg/pull/56649#issuecomment-2024383614 (clarify that the calls are only used for CSS validation, not output of CSS).
* Update the classname used to make it clearer that the classname will not be used in any output.

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a post or page. -->
<!-- 2. Insert a heading block. -->
<!-- 3. etc. -->

1. Proofreading that what's written is accurate 🙂
2. To be sure this doesn't cause any issues, in global styles add Additional CSS, and enter it in an malformed way. Blur the field and check that the error message still appears. Then, go to fix the error in the CSS, and check that once it's resolved, the error disappears.
3. Do the same at the block-level in global styles. E.g. go to the Group block in global styles, and enter additional CSS under the Advanced panel. Example illegal CSS: `asdf`, example legal CSS: `border: 1px solid;`

### Testing Instructions for Keyboard
<!-- How can you test the changes by using the keyboard only? Please note, this is required for PRs that change the user interface (UI). This ensures the PR can be tested for any possible accessibility regressions. -->

## Screenshots or screencast <!-- if applicable -->

andrewserong committed:
Additional CSS: Add code comments contextualising tranformStyles for clarity

tellthemachines commented on packages/block-editor/src/components/global-styles/advanced-panel.js:
```diff
@@ -28,6 +28,10 @@ export default function AdvancedPanel( {
 		if ( cssError ) {
+			// Pass a wrapping selector to transformStyles to ensure that the CSS
+			// tested resembles actual CSS rules. This is necessary because the additional CSS
+			// field supports CSS rules that are not wrapped in a selector.
```
To be precise, we need to pass a wrapper selector because otherwise the transformStyle function will [return before it does the error checking](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/utils/transform-styles/index.js#L18). I don't think it really matters whether the actual CSS is a whole rule or just a property or two. Afaict inputting a single property will output the same (unwrapped) property, whether we pass a wrapper or not.

It would probably be better to have a dedicated error checking function instead of using this one 🤔  That might be a good improvement to add to the list for additional CSS!

aaronrobertshaw reviewed (commented):
Thanks for following up on this @andrewserong 👍 

As these comments note, the selector is only for validation purposes. I think we could make this obvious by replacing `.editor-styles-wrapper` here with something like `.for-validation-only`.

What do you think?

andrewserong committed:
Simplify comments, use a classname that more obviously flags that the classname will not be used in output

andrewserong commented:
Thanks for the feedback and good ideas, folks! I've updated the classname strings to `.for-validation-only` and simplified the comments a little. Let me know if you'd like any other tweaks.

aaronrobertshaw reviewed (approved):
Thanks for the quick tweaks 🚀 

LGTM 🚢
